airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java

```diff
@@ -1354,16 +1372,13 @@ private static ConnectionState waitForConnectionState(final AirbyteApiClient api
   }
 
   private static void waitForTemporalWorkflow(final UUID connectionId) {
-    /*
-     * do { try { Thread.sleep(1000); } catch (InterruptedException e) { throw new RuntimeException(e);
-     * } } while
-     * (temporalClient.isWorkflowRunning(temporalClient.getConnectionManagerName(connectionId)));
-     */
-    try {
-      Thread.sleep(10 * 1000);
-    } catch (InterruptedException e) {
-      throw new RuntimeException(e);
-    }
+    do {
+      try {
+        Thread.sleep(1000);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+    } while (!temporalClient.isWorkflowRunning(temporalClient.getConnectionManagerName(connectionId)));
```

Comment:
Can we add a timeout here directly or a junit `Timeout`? to everything calling this function?

Reply:
Done